### PR TITLE
Fixed telnet LS commond throw class cast exception

### PR DESCRIPTION
**Pull request by zcbbpo**
Fixed for [telnet for ls commond](https://github.com/apache/dubbo/issues/9765)
1. modify dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/command/util/ServiceCheckUtils.java
2. add unit test at dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by BurningCN** [comment]
Hello, I want to confirm that this problem appeared in `master` branch, right? `3.0` branch has fixed this problem (`ServiceDiscoveryRegistry extends FailbackRegistry extends AbstractRegistry`), However, it is also an optimization to make a type judgment before the forced cast.

File: dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java [hunk 1]
```diff
@@ -17,19 +17,25 @@
...
 import org.apache.dubbo.qos.DemoServiceImpl;
-import org.apache.dubbo.rpc.model.ApplicationModel;
-import org.apache.dubbo.rpc.model.ModuleServiceRepository;
-import org.apache.dubbo.rpc.model.ProviderModel;
-import org.apache.dubbo.rpc.model.ServiceDescriptor;
-import org.apache.dubbo.rpc.model.ServiceMetadata;
+import org.apache.dubbo.registry.Registry;
+import org.apache.dubbo.registry.client.ServiceDiscoveryRegistry;
+import org.apache.dubbo.registry.support.RegistryManager;
+import org.apache.dubbo.rpc.model.*;
```
Comment (BurningCN) [inline dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java:29]: don't use *

Comment (zcbbpo) [reply dubbo-plugin/dubbo-qos/src/test/java/org/apache/dubbo/qos/command/util/ServiceCheckUtilsTest.java]: already modified.

**Comment by zcbbpo** [comment]
Hi BurningCN
you are right. For 3.0 version, it's extend from AbstractRegistry.
because something , we are project must use 2.7.13. so we need do this.

**Comment by zcbbpo** [comment]
Do i should close this pr?

**Comment by chickenlj** [comment]
Please submit a similar one against the master branch

**Pull request closed without merging**